controllers/volumereplicationgroup_controller.go

```diff
@@ -920,7 +984,29 @@ func (v *VRGInstance) handleVRGMode(state ramendrv1alpha1.ReplicationState) bool
 		}
 	}
 
-	return asyncNeedRequeue
+	if v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled {
+		// mark all PVCs as protected.
+		v.markAllPVCsProtected()
```

Comment:
(future) This should called only if `asyncNeedRequeue` is false, so that we do not mark PVCs that need async mode protection as ready due to sync protection being active.

Reply:
Future, because currently we are not supporting dual mode for the workload.